src/Simulation/Enrollment.jl

```diff
@@ -0,0 +1,142 @@
+
+                    # Enroll in coreqs
+                    for coreqId in coreqIds
+                        coreq = getCourseById(courses, coreqId)
+
+                        if canEnroll(student, coreq, courses, studentProgress, max_credits, currentTerm)
+                            # Enroll the student in the course
+                            push!(coreq.metadata["students"], student)
+
+                            # Increment the course's enrollment counters
+                            coreq.metadata["enrolled"] += 1
+                            coreq.metadata["termenrollment"][currentTerm] += 1
+
+                            # Increse the student's term credits
+                            student.termcredits += coreq.credit_hours
+                        end
+                    end
+
+
+
+                    # Determine wheter the student can be enrolled in the current course.
+                    if canEnroll(student, course, courses, studentProgress, max_credits, currentTerm)
+
+                        # Enroll the student in the course
+                        push!(course.metadata["students"], student)
+
+                        # Increment the course's enrollment counters
+                        course.metadata["enrolled"] += 1
+                        course.metadata["termenrollment"][currentTerm] += 1
+
+                        # Increse the student's term credits
+                        student.termcredits += course.credit_hours
+                    end
+                end
+            end
+        end
+    end
+
+    # Function that determines wheter a student can enroll in a course
+    function canEnroll(student, course, courses, studentProgress, max_credits, term)
+        # Find the prereq ids of the current course
+        prereqs = getReqs(courses, course, pre)
+        prereqIds = map(x -> x.metadata["id"], prereqs)
+
+        !in(student, course.metadata["students"]) &&
+            (length(prereqIds) == 0 || sum(studentProgress[student.id, prereqIds]) == length(prereqIds)) &&     # No Prereqs or the student has completed them
+            studentProgress[student.id, course.metadata["id"]] == 0.0 &&                                        # The student has not already completed the course
+            !isStudentEnrolled(student, course) &&                                                              # The student has not already enrolled the course
+            student.termcredits + course.credit_hours <= max_credits &&                                         # The student will not exceed the maximum number of credit hours
+            course.metadata["termReq"] <= term &&                                                               # The student must wait until the term req has been met
+            enrolledInCoreqs(student, course, courses, studentProgress)                                         # The student is enrolled in or has completed coreqs
```

Comment:
Does this implicitly return true or false since it's the last statement in the function? Might not hurt to explicitly return?

Reply:
Done. 3f2a7dc0a7d671505b2c7a0ab9b9e6c76fea3a55